warehouse/packaging/models.py

```diff
@@ -194,6 +197,35 @@ def documentation_url(self):
 
         return request.route_url("legacy.docs", project=self.name)
 
+    @property
+    def all_releases(self):
+        return sorted(
+            orm.object_session(self).query(Release)
+            .filter(Release.project == self)
+            .options(orm.load_only(
```

Comment:
Having ``load_only`` only a "generic" name like ``all_releases`` or ``latest_release`` makes me nervous because I feel like it's very easy to trigger N+1 query problems  if this gets used in another context without realizing it, since the name doesn't imply anything about the fact it's limiting the data available.

What information do we actually use out of these? I don't remember what the templates do, but it looks like the search indexing really only cares about version numbers, is it the same for the templates? If it is maybe something like:


```
@property
def all_versions(self):
    return (orm.object_session(self)
               .query(Release.version)
               .filter(Release.project == self)
               .order_by(Release._pypi_ordering.desc())
               .all())
```


Would be enough to do the trick? We'd probably want a ``list()`` or something in there so the return type is just ``["1.0", "2.0", 3.0"]`` in whatever order we expect this in.


Another thing to think about is making these just functions that aren't attached to the ``Project`` object, I'm not sure if that ends up being clearer or not.

Reply:
forklift/legacy.py also needs to pull in all_releases to re enumerate `_pypi_ordering`. I guess it'd be OK to do what's proposed here and inline the query for that.